src/main/kotlin/com/ampnet/blockchainapiservice/service/VerificationServiceImpl.kt

```diff
@@ -29,4 +51,69 @@ class VerificationServiceImpl(
 
         return unsignedVerificationMessageRepository.store(message)
     }
+
+    override fun verifyMessageSignature(messageId: UUID, signature: String): SignedVerificationMessage {
```

Comment:
maybe change the name of the function?
_verifyMessageSignature()_ implies a stateless function but this one also stores the data if verification successful.

_verifyAndStoreMessageSignature()_ is a bit long but communicates the function implementation more clearly